src/react/components/common/assetPreview/tfrecordAsset.tsx

```diff
@@ -0,0 +1,84 @@
+import React from "react";
+import { IAssetProps } from "./assetPreview";
+import { IAsset, AssetType } from "../../../../models/applicationState";
+import HtmlFileReader from "../../../../common/htmlFileReader";
+import { TFRecordsReader } from "../../../../providers/export/tensorFlowRecords/tensorFlowReader";
+import { FeatureType } from "../../../../providers/export/tensorFlowRecords/tensorFlowBuilder";
+
+/**
+ * Properties for TFRecord Asset Image component
+ */
+export interface ITFRecordProps extends IAssetProps, React.ClassAttributes<TFRecordAsset> {
+}
+
+/**
+ * State for TFRecord Asset Image component
+ * @member tfRecordImage64 - base64 representation of the image data
+ */
+export interface ITFRecordState {
+    tfRecordImage64: string;
+}
+
+/**
+ * React component that displays an image from a TFRecord asset file
+ */
+export class TFRecordAsset extends React.Component<ITFRecordProps, ITFRecordState> {
+    private image: React.RefObject<HTMLImageElement> = React.createRef();
+
+    constructor(props, context) {
```

Comment:
In our new components let avoid creating a constructor unless really needed.  You can define the state of the component inline within the class like
```
public state: ITFRecordState = {
    tfRecordImage64: "",
};
```

Reply:
Then you can ommit the constructor entirely.